packages/react-signer/src/Modal.tsx

```diff
@@ -146,12 +167,16 @@ class Signer extends React.PureComponent<Props, State> {
     };
   }
 
-  public async componentDidUpdate (): Promise<void> {
+  public async componentDidUpdate (_: Props, prevState: State): Promise<void> {
     const { currentItem } = this.state;
 
     if (currentItem && currentItem.status === 'queued' && !(currentItem.extrinsic || currentItem.payload)) {
       return this.sendRpc(currentItem);
     }
+
+    if (currentItem?.isSign && currentItem?.accountId && currentItem.accountId !== prevState.currentItem?.accountId) {
```

Comment:
Would probably not do this. Rather just rely on the `api` (which also needs to re-enable this one https://github.com/polkadot-js/api/pull/1525)

Reply:
Actually, the ability to edit the nonce is very useful - if you e.g. sign 3 transactions one after another then they need to have different nonces for each of them to work once submitted, so I wouldn't just automatically use the current nonce from the api.